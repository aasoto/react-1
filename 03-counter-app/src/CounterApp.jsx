import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => {

  const [ counter, setCounter ] = useState( value );

  const handleAdd = () => {
    setCounter((c) => {
      return c + 1;
    })
  }

  const handleSubtract = () => {
    setCounter((c) => {
      return c - 1;
    })
  }

  const handleReset = () => {
    setCounter(value)
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2 aria-label='counter'> { counter } </h2>

      <button aria-label='increment' onClick={ handleAdd }> +1 </button>
      <button aria-label='decrement' onClick={ handleSubtract }> -1 </button>
      <button aria-label='reset' onClick={ handleReset }> Reset </button>

    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}