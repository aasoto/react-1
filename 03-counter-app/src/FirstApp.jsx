const newMessage = {
  message: 'Bienvenidos a mi primera aplicación.',
  title: ' Andrés Soto React'
}

const getMessage = () => {
  return newMessage.message;
}

const getTitle = () => {
  return newMessage.title;
}

const sumar = (a, b) => {
  return a + b;
}

export const FirstApp = () => {
  /**
   * Para agrupar un numero de elementos debe haber un nodo padre,
   * el cual se puede definir con la etiqueta <Fragment></Fragment>
   * de manera resumida <></>
   */
  return (
    <>
      <h1>{getTitle()}</h1>
      <h3>{sumar(5, 3)}</h3>
    </>
  );
}