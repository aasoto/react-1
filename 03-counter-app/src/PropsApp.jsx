import PropTypes from 'prop-types'

export const PropsApp = ({title, subTitle, developper}) => {
  return (
    <>
      <a href="#">
        <h1 data-testid="test-title">{title}</h1>
      </a>
      <h4>{subTitle}</h4>
      <h4>{subTitle}</h4>
      <p>Desarrollador por: {developper}</p>
    </>
  );
}

PropsApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  developper: PropTypes.string
}

PropsApp.defaultProps = {
  developper: 'Andrés Soto',
  subTitle: 'No hay subtítulo',
}