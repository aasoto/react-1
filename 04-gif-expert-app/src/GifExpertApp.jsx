import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One Punch', 'Dragon Ball'])

  const onAddCategory = () => {
    setCategories(['Valorant', ...categories]);
  }

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory setCategories={setCategories}/>
      <button onClick={onAddCategory}>Agregar categor√≠a</button>
      <ol>
        {
          categories.map(category => {
            return <li key={category}>{category}</li>
          })
        }
      </ol>
    </>
  );
}