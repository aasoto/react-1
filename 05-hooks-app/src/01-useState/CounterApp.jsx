import { useState } from "react"
import { Fragment } from "react"

export const CounterApp = () => {

  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const {counter1, counter2, counter3} = counter;

  return (
    <Fragment>
      <h1>Counter App</h1>
      <hr />
      <h2>Counter 1: {counter1}</h2>
      <h2>Counter 2: {counter2}</h2>
      <h2>Counter 3: {counter3}</h2>
      <hr />
      <button className="btn btn-primary" onClick={() => setCounter(
        {
          ...counter,
          counter1: counter1 + 1,
        }
        )}>+1</button>
    </Fragment>
  )
}
