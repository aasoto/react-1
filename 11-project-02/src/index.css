@tailwind base;
@tailwind components;
@tailwind utilities;

.navbar {
  @apply fixed z-20 flex justify-center items-center w-full h-24 p-4
}

.navbar nav{
  @apply w-full h-full backdrop-blur-md bg-white/50 rounded-lg px-5 md:px-10 py-2 shadow-lg
}

.navbar .nav-container{
  @apply flex justify-between items-center
}

.navbar .nav-section{
  @apply flex items-center gap-5
}

.navbar h1{
  @apply text-black font-bold text-2xl italic
}