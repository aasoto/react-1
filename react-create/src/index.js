// Variables y constantes

const nombre = 'AndrÃ©s';
let