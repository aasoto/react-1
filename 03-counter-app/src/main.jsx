import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import HelloWorldApp from './HelloWorld';
import { PropsApp } from './PropsApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <PropsApp title="Probando PropTypes" subTitle='Esto es un subtÃ­tulo'/>
    {/* <CounterApp value={4} /> */}
  </React.StrictMode>
);