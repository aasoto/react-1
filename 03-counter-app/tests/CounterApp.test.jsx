import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas de <CounterApp />', () => {

  const start = 100;

  test('hacer match con el snapshot', () => {
    const {container} = render(<CounterApp value={start}/>);
    expect(container).toMatchSnapshot();
  });

  test('mostrar valor iniciar de 100', () => {
    render(<CounterApp value={start}/>);
    expect(parseInt(screen.getByRole('heading', {level: 2}).innerHTML)).toBe(start);
  });

  test('debe de incrementar con el botón +1', () => {
    render(<CounterApp value={start}/>);
    fireEvent.click(screen.getByRole('button', {name: 'increment'}));
    expect(parseInt(screen.getByRole('heading', {level: 2, name: 'counter'}).innerHTML)).toBe(start + 1);
  });

  test('debe de incrementar con el botón -1', () => {
    render(<CounterApp value={start}/>);
    fireEvent.click(screen.getByRole('button', {name: 'decrement'}));
    expect(parseInt(screen.getByRole('heading', {level: 2, name: 'counter'}).innerHTML)).toBe(start - 1);
  });

  test('debe regresar el conteo al valor inicial', () => {
    render(<CounterApp value={start}/>);

    fireEvent.click(screen.getByRole('button', {name: 'increment'}));
    fireEvent.click(screen.getByRole('button', {name: 'increment'}));
    fireEvent.click(screen.getByRole('button', {name: 'increment'}));

    fireEvent.click(screen.getByRole('button', {name: 'reset'}));

    expect(parseInt(screen.getByRole('heading', {level: 2, name: 'counter'}).innerHTML)).toBe(start);
  });
});