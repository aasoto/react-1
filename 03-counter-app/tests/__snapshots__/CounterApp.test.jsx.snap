// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <CounterApp /> hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    aria-label="counter"
  >
     
    100
     
  </h2>
  <button
    aria-label="increment"
  >
     +1 
  </button>
  <button
    aria-label="decrement"
  >
     -1 
  </button>
  <button
    aria-label="reset"
  >
     Reset 
  </button>
</div>
`;
